Documentation/policy/troubleshooting.rst

```diff
@@ -166,101 +168,109 @@ In the above example, for one of the ``deathstar`` pods the endpoint id is 568.
+            "any:class": "mediabot",
+            "any:org": "empire",
+            "k8s:io.kubernetes.pod.namespace": "default"
+          }
+        },
+        "egress": [
+          {
+            "toFQDNs": [
+              {
+                "matchName": "api.twitter.com"
+              }
+            ]
+          },
+          {
+            "toEndpoints": [
+              {
+                "matchLabels": {
+                  "k8s:io.kubernetes.pod.namespace": "kube-system",
+                  "k8s:k8s-app": "kube-dns"
+                }
+              }
+            ],
+            "toPorts": [
+              {
+                "ports": [
+                  {
+                    "port": "53",
+                    "protocol": "ANY"
+                  }
+                ],
+                "rules": {
+                  "dns": [
                     {
-                      "derived-from-rules": [
-                        [
-                          "k8s:io.cilium.k8s.policy.name=rebel-escape",
-                          "k8s:io.cilium.k8s.policy.uid=c96f66a8-135e-11e9-babd-080027d2d952",
-                          "k8s:io.cilium.k8s.policy.namespace=default",
-                          "k8s:io.cilium.k8s.policy.derived-from=CiliumNetworkPolicy",
-                          "cilium-generated:ToFQDN-UUID=9a1d4006-1360-11e9-a6d4-080027d2d952"
-                        ]
-                      ],
-                      "rule": "104.198.14.52/32"
+                      "matchPattern": "*"
                     }
-                  ],
-                  "ingress": []
-                },
+                  ]
+                }
+              }
+            ]
+          }
+        ],
+        "labels": [
+          {
+            "key": "io.cilium.k8s.policy.derived-from",
+            "value": "CiliumNetworkPolicy",
+            "source": "k8s"
+          },
+          {
+            "key": "io.cilium.k8s.policy.name",
+            "value": "fqdn",
+            "source": "k8s"
+          },
+          {
+            "key": "io.cilium.k8s.policy.namespace",
+            "value": "default",
+            "source": "k8s"
+          },
+          {
+            "key": "io.cilium.k8s.policy.uid",
+            "value": "fc9d6022-2ffa-4f72-b59e-b9067c3cfecf",
+            "source": "k8s"
+          }
+        ]
+      }
+
+
+#. After making a DNS request, the FQDN to IP mapping should be available via
+   ``cilium fqdn cache list``:
+
+   .. code-block:: shell-session
+
+      # cilium fqdn cache list
+      Endpoint   FQDN                TTL      ExpirationTime             IPs
+      2761       help.twitter.com.   604800   2019-07-16T17:57:38.179Z   104.244.42.67,104.244.42.195,104.244.42.3,104.244.42.131
+      2761       api.twitter.com.    604800   2019-07-16T18:11:38.627Z   104.244.42.194,104.244.42.130,104.244.42.66,104.244.42.2
+
+#. If the traffic is allowed, then these IPs should have corresponding local identities via
```

Comment:
I was actually pleasantly surprised that we don't actually allocate local CIDR identities for these FQDNs if there is no policy corresponding to them, but it makes sense and means we have a bunch less work to do (for example, we don't need to evaluate policy against CIDRs that are always denied when they aren't selected by any FQDN selector)

Reply:
Yup, exactly :D